tests/Feature/Traits/CatchFailures.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Tests\Feature\Traits;
+
+use Illuminate\Testing\TestResponse;
+
+trait CatchFailures
+{
+	protected function assertStatus(TestResponse $response, int $expectedStatusCode): void
+	{
+		if ($response->getStatusCode() === 500) {
+			$exception = $response->json();
+			$this->trimException($exception);
+			// $exception['trace'] = array_slice($exception['trace'], 0 ,3);
+			dump($exception);
+		}
+		$response->assertStatus($expectedStatusCode);
+	}
+
+	private function trimException(array &$exception): void
+	{
+		$exception['trace'] = array_slice($exception['trace'], 0, 3);
```

Comment:
I assume, but this is more of a guess, you want to have the named array keys `'file'`, `'line'` and `'function'` of the array which happens to be the first three entries of the array most of the time. But that is not guaranteed. If this is what you want, make it explicit
```suggestion
		$exception['trace'] = array_intersect_key($exception['trace'], array_flip(['file', 'line', 'function']));
```

Reply:
No, this trims down the exception thrown to the last 3 calls of the stack trace rather then 78 and making it pretty unreadable.
e.g. :
```
.array:6 [ // tests/Feature/Traits/CatchFailures.php:14
  "message" => "Class "App\DTO\AlbumProtectionPolicies" not found"
  "exception" => "Error"
  "file" => "/home/biv/Documents/Lychee/app/SmartAlbums/BaseSmartAlbum.php"
  "line" => 151
  "trace" => array:3 [
    0 => array:5 [
      "file" => "/home/biv/Documents/Lychee/app/DTO/AbstractAlbumDTO.php"
      "line" => 29
      "function" => "toArray"
      "class" => "App\SmartAlbums\BaseSmartAlbum"
      "type" => "->"
    ]
    1 => array:5 [
      "file" => "/home/biv/Documents/Lychee/app/DTO/DTO.php"
      "line" => 64
      "function" => "toArray"
      "class" => "App\DTO\AbstractAlbumDTO"
      "type" => "->"
    ]
    2 => array:5 [
      "file" => "/home/biv/Documents/Lychee/app/DTO/DTO.php"
      "line" => 41
      "function" => "jsonSerialize"
      "class" => "App\DTO\DTO"
      "type" => "->"
    ]
  ]
  "previous_exception" => null
]
```